### PR TITLE
Fix progress model and validator

### PR DESCRIPTION
**Pull request by DaleDMT**

### CODE REVIEW

File: ckanext/dgua/controllers/progress.py [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+    def new(self):
+        redirect_if_not_sysadmin(c.userobj)
+        form = forms.ProgressOrganizationNewForm(tk.request.POST)
+        if not tk.request.POST:
+            return tk.render('progress/new_organization.html', {'form': form})
+        if form.validate():
+            organization = ProgressOrganization(name=form.organization_name.data, metric=form.metric.data,
+                                                is_show=form.is_show.data)
+            organization.save()
+            h.flash_success(tk._(u'Organization {} successfully added'.format(organization.name)))
```
Comment (Krokop) [inline ckanext/dgua/controllers/progress.py]: 1. Need to change format method to %
2. Move template and content on separate lines.

File: ckanext/dgua/controllers/progress.py [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+
+    def edit(self, organization_id):
+        redirect_if_not_sysadmin(c.userobj)
+        form = forms.ProgressOrganizationEditForm(tk.request.POST)
+        organization = get_valid_organization(organization_id)
+        if tk.request.POST:
+            if form.validate():
+                organization.update(new_name=form.organization_name.data, new_metric=form.metric.data,
+                                    is_show=form.is_show.data)
+                h.flash_success(tk._(u'Organization {} successfully changed'.format(organization.name)))
```
Comment (Krokop) [inline ckanext/dgua/controllers/progress.py]: same here

File: ckanext/dgua/controllers/progress.py [hunk 3]
```diff
@@ -0,0 +1,143 @@
...
+        form.is_show.data = organization.is_show
+
+        return tk.render('progress/edit_organization.html',
+                         {'form': form, 'organization_list': ProgressOrganization.all(),
+                          'organization_id': organization.id})
+    def delete(self, organization_id):
+        redirect_if_not_sysadmin(c.userobj)
+        organization = get_valid_organization(organization_id)
+        organization.delete()
+        h.flash_success(tk._(u'Organization {} was successfully deleted'.format(organization.name)))
```
Comment (Krokop) [inline ckanext/dgua/controllers/progress.py]: same here

File: ckanext/dgua/controllers/progress.py [hunk 4]
```diff
@@ -0,0 +1,143 @@
...
+        organization_list = ProgressOrganization.all()
+        if not tk.request.POST:
+            return tk.render('progress/new_package.html', {'form': form, 'organization_list': organization_list})
+        if form.validate():
+            package = ProgressPackage(name=form.name.data,
+                                      organization_id=int(form.organization_id.data),
+                                      url=form.url.data,
+                                      qa=form.qa.data)
+            package.save()
+            h.flash_success(tk._(u'Package {} successfully added'.format(package.name)))
```
Comment (Krokop) [inline ckanext/dgua/controllers/progress.py]: same here

File: ckanext/dgua/controllers/progress.py [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+
+    def edit(self, package_id):
+        redirect_if_not_sysadmin(c.userobj)
+        form = forms.ProgressPackageEditForm(tk.request.POST)
+        package = get_valid_package(package_id)
+        if tk.request.POST:
+            if form.validate():
+                package.update(name=form.name.data, organization_id=form.organization_id.data,
+                               url=form.url.data, qa=form.qa.data)
+                h.flash_success(tk._(u'Package {} successfully changed'.format(package.name)))
```
Comment (Krokop) [inline ckanext/dgua/controllers/progress.py]: same here

File: ckanext/dgua/controllers/progress.py [hunk 6]
```diff
@@ -0,0 +1,143 @@
...
+
+        return tk.render('progress/edit_package.html', {'form': form,
+                                                        'organization_list': ProgressOrganization.all(),
+                                                        'package_id': package.id})
+
+    def delete(self, package_id):
+        redirect_if_not_sysadmin(c.userobj)
+        package = get_valid_package(package_id)
+        package.delete()
+        h.flash_success(tk._(u'Package {} was successfully deleted'.format(package.name)))
```
Comment (Krokop) [inline ckanext/dgua/controllers/progress.py]: same here

File: ckanext/dgua/models.py [hunk 7]
```diff
@@ -469,20 +469,20 @@ def delete(self, commit=True):
     })
 
 
-in_progress_organization_table = Table('in_progress_organization', meta.metadata,
```
Comment (Krokop) [inline ckanext/dgua/models.py]: Need to write migration

File: ckanext/dgua/models.py [hunk 8]
```diff
@@ -535,29 +534,31 @@ def delete(self, commit=True):
...
-                                  Column('id', types.Integer, primary_key=True, autoincrement=True),
-                                  Column('organization_id', types.Integer,
-                                         ForeignKey('in_progress_organization.id',
-                                                    onupdate='CASCADE',
-                                                    ondelete='CASCADE'), nullable=False, index=True),
-                                  Column('name', types.Unicode(255), nullable=False),
-                                  Column('url', types.Unicode(), nullable=True),
-                                  UniqueConstraint('name')
-                                  )
+progress_package_table = Table('progress_package', meta.metadata,
```
Comment (Krokop) [inline ckanext/dgua/models.py]: Need to write migration

File: ckanext/dgua/controllers/progress.py (hunk 2 above)
Comment (DaleDMT) [reply ckanext/dgua/controllers/progress.py]: Done

File: ckanext/dgua/controllers/progress.py (hunk 3 above)
Comment (DaleDMT) [reply ckanext/dgua/controllers/progress.py]: Done

File: ckanext/dgua/controllers/progress.py (hunk 4 above)
Comment (DaleDMT) [reply ckanext/dgua/controllers/progress.py]: Done

File: ckanext/dgua/controllers/progress.py (hunk 5 above)
Comment (DaleDMT) [reply ckanext/dgua/controllers/progress.py]: Done

File: ckanext/dgua/controllers/progress.py (hunk 6 above)
Comment (DaleDMT) [reply ckanext/dgua/controllers/progress.py]: Done

**Comment by e0ne** [comment]
LGTM but @Krokop's comments should be addressed

**Comment by Krokop** [comment]
@DaleDMT please add prefix DGUA in future for models, forms, controls. https://github.com/908Inc/ckanext-dgua/pull/147/files#diff-85e96d0b5d9fbfbb79b3deb655313065R636

**Comment by DaleDMT** [comment]
OK, I will make in a separate PR

**Pull request merged**